presto-main/src/test/java/com/facebook/presto/type/TestTimestamp.java

```diff
@@ -237,6 +243,49 @@ public void testCastFromSlice()
     }
 
     @Test
+    public void testCastFromSliceFailsOnTimestampWithTimeZone()
+    {
+        if (connectorSession().isLegacyTimestamp()) {
+            // we expect those not to fail in legacy mode
+            return;
```

Comment:
or `throw new SkipException` if you like..  this is psychological problem of course: to choose from

* "lie that that passed while it did not test anything"
* "like we have skipped tests while we did what we could not to"